### PR TITLE
Theme, Documentation and Weblog 

### PR DESCRIPTION
**Pull request by Kappuccino111**
Completes the initial setup of the documentation @jonas-rem . Please review and merge.

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -9,96 +9,79 @@
...
-# so a file named "default.css" will overwrite the builtin "default.css".
-# html_static_path = ['_static']
-
-# -- Options for PDF output -------------------------------------------------
-
 latex_elements = {
-    'fontpkg': '\\usepackage{lmodern}',
-    'papersize': 'a4paper',
-    'extraclassoptions': 'oneside',
+    'papersize': 'letterpaper',
```
Comment (jonas-rem) [inline doc/source/conf.py:75]: A4 is more common, right? I think only North America uses letter paper format.

File: doc/source/conf.py [hunk 2]
```diff
@@ -9,96 +9,79 @@
...
 html_context = {
-    "display_github": True,
-    "github_repo": "jonas-rem/lwm2m_server",
-    "github_version": "main",
-    "conf_py_path": "/doc/source/",
+    "display_github": True,  
+    "github_user": "jonas-rem",  
+    "github_repo": "lwm2m_server",  
+    "github_version": "main", 
+    "conf_py_path": "/docs/source/",
```
Comment (jonas-rem) [inline doc/source/conf.py:71]: The whole html_context block is now handled in the theme options. As far as I saw we can remove this block entirely now.

File: doc/source/conf.py [hunk 3]
```diff
@@ -9,96 +9,79 @@
...
+    version = "0.1"  
+release = version
 
-# Add any Sphinx extension module names here, as strings. They can be
-# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
-# ones.
 extensions = [
     'sphinx.ext.autodoc',
-    'sphinxcontrib.plantuml',
-    'sphinx.ext.imgconverter'
```
Comment (jonas-rem) [inline doc/source/conf.py:36]: The imageconverter package is required for converting svg images for the pdf/latex output.

File: doc/source/conf.py [hunk 4]
```diff
@@ -9,96 +9,79 @@
...
-
-# Add any paths that contain custom static files (such as style sheets) here,
-# relative to this directory. They are copied after the builtin static files,
-# so a file named "default.css" will overwrite the builtin "default.css".
-# html_static_path = ['_static']
-
-# -- Options for PDF output -------------------------------------------------
-
 latex_elements = {
-    'fontpkg': '\\usepackage{lmodern}',
```
Comment (jonas-rem) [inline doc/source/conf.py:76]: I think using lmodern font makes the appearance of the generated pdf much better.

File: doc/source/index.rst [hunk 5]
```diff
@@ -1,8 +1,28 @@
...
+=========================================
 
+.. image:: _static/logo.jpeg
+   :alt: Flow Nexus Logo
+   :class: logo
+
+Introduction
+------------
+
+Welcome to Flow Nexus, your comprehensive solution for small to medium scale IoT deployments. This platform is designed to streamline the setup, configuration, and use of IoT devices, making it easier than ever to manage your IoT ecosystem.
```
Comment (jonas-rem) [inline doc/source/index.rst:11]: Now comes the linewidth discussion ;). I prefer 80 chars per line (80-100 in c-code, max 120 in python/java code) as this improves readability in the editor. Do you have a particular preference?

Comment (jonas-rem) [inline doc/source/index.rst:11]: s/easier than ever/easy

For the documentation we should focus on a scientific writing style whenever possible, avoiding adjectives and phrases that represent emotion. This is the way papers are written, it represents more seriousness.

File: doc/source/index.rst [hunk 6]
```diff
@@ -1,8 +1,28 @@
...
+
+Motivation
+----------
+
+This project aims to provide a minimal and standalone Open Source IoT platform.The focus is on systems with a deployment of up to a few thousand devices, it does not aim to be scalable to millions of devices. The goal is to build a system that recieves e.g. regular temperature samples from remote IoT devices. The temperature samples will be send to a backend and stored in a database. The data is be visualized in a web application.
+
+Vision
+------
+Our vision for Flow Nexus is to create a user-friendly, efficient, and robust platform for IoT management. We aim to simplify the complexities of IoT deployments, making them accessible and manageable for businesses of all sizes.
```
Comment (jonas-rem) [inline doc/source/index.rst:20]: should we add `Open Source`?

**Review by jonas-rem — CHANGES_REQUESTED** [review]
Prompt highlighting does not work anymore with this theme, but I could not figure out why. But that is not so important right now, we can look into it later.

This should usually highlight the prompt in a different color do distinguish between prompt and command.
```
.. code-block:: console

  host:lwm2m_server/server$ cmd
```

The logo looks good in the dark mode, but not really good in the light mode. Is it possible to make the background of the image transparent so that it looks good in any environment?

**Comment by jonas-rem** [comment]
@Kappuccino111 the theme looks good, I like the dynamic rendering of the contents section on the right! The Weblog is also a good idea to inform about the project. A more detailed/technical info will anyways be part of releases.

File: doc/requirements.txt [hunk 7]
```diff
@@ -1,5 +1,7 @@
 Sphinx==7.3.7
-sphinx-rtd-theme==2.0.0
+sphinx-book-theme
```
[the remark this answers is not in the document]
Comment (Kappuccino111) [reply doc/requirements.txt]: - [x] Mentioned the theme versioning 
- [x] Wrote a custom css file for highlighting the prompt.
- [x] Modified the logo to be more artistic and also support transparent PNGs. Looks much better now 😁

File: doc/source/conf.py [hunk 8]
```diff
@@ -9,96 +9,79 @@
...
-version = re.sub('', '', os.popen('git describe --tags').read().strip())
+project = 'Flow Nexus'
+copyright = '2024, Jonas Remmert, Akarshan Kapoor'
+author = 'Jonas Remmert, Akarshan Kapoor'
 
-# -- General configuration ---------------------------------------------------
+try:
+    version = subprocess.check_output(["git", "describe", "--tags"]).decode().strip()
+except subprocess.CalledProcessError:
+    version = "0.1"
```
[the remark this answers is not in the document]
Comment (Kappuccino111) [reply doc/source/conf.py]: - [x] Removed trailing whitespaces 
- [x] Removed the try catch block (did not have git tags configured earlier, will need a note in the readme for the same?)

File: doc/source/conf.py [hunk 9]
```diff
@@ -9,96 +9,79 @@
...
-    'sphinxcontrib.plantuml',
-    'sphinx.ext.imgconverter'
+    'sphinx.ext.viewcode',
+    'sphinx.ext.napoleon',  
+    'sphinxcontrib.plantuml',  
+    'sphinx.ext.todo',
+    'sphinx.ext.coverage',
+    'sphinx.ext.ifconfig',
+    'sphinx.ext.githubpages', 
+    'sphinxcontrib.newsfeed',
```
[the remark this answers is not in the document]
Comment (Kappuccino111) [reply doc/source/conf.py]: - [x] Was trying to configure multiple things, but would have been too complicated. So removed them for now 👍

File: doc/source/conf.py [hunk 10]
```diff
@@ -9,96 +9,79 @@
...
+    'repository_branch': 'main',
+    'path_to_docs': 'docs/source/',
+    'use_repository_button': True,
+    'use_issues_button': True,
+    'use_edit_page_button': True,
+    'use_download_button': True,
+    'show_navbar_depth': 100,
+    'home_page_in_toc': True,
+    'navigation_with_keys': False,
+    "announcement": "🚧 This repository is currently in active development. Some features may not be fully functional or may change significantly.🚧",
```
[the remark this answers is not in the document]
Comment (Kappuccino111) [reply doc/source/conf.py]: - [x] Yep, I also had that in my mind initially 😂. Removed.

File: doc/source/index.rst (hunk 5 above)
Comment (Kappuccino111) [reply doc/source/index.rst]: - [x] Fixed for better readability

File: doc/source/conf.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (Kappuccino111) [reply doc/source/conf.py]: @jonas-rem there seems to be the same issue in the CI pipeline, failing the git tag check

File: doc/source/conf.py (hunk 3 above)
Comment (Kappuccino111) [reply doc/source/conf.py:36]: - [x] Yep, added back again.

**Comment by Kappuccino111** [comment]
@jonas-rem have updated most of the changes. Please check and merge

**Pull request closed without merging**